src/Analyzers/CSharp/Tests/ImplementInterface/ImplementInterfaceTests.cs

```diff
@@ -11976,4 +12011,122 @@ static explicit I11<C11>.operator int(C11 x)
             CodeActionVerifier = (codeAction, verifier) => verifier.Equal(CodeFixesResources.Implement_all_members_explicitly, codeAction.Title),
         }.RunAsync();
     }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/67023")]
+    public async Task TestIEnumerable1()
+    {
+        await TestWithAllCodeStyleOptionsOffAsync(
+            """
+            using System;
+            using System.Collections;
+            using System.Collections.Generic;
+
+            class Class : {|CS0535:{|CS0535:IEnumerable<int>|}|}
+            {
+            }
+            """,
+            """
+            using System;
+            using System.Collections;
+            using System.Collections.Generic;
+
+            class Class : IEnumerable<int>
+            {
+                public IEnumerator<int> GetEnumerator()
+                {
+                    throw new NotImplementedException();
+                }
+            
+                IEnumerator IEnumerable.GetEnumerator()
+                {
+                    return GetEnumerator();
+                }
+            }
+            """);
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/67023")]
+    public async Task TestIEnumerable2()
+    {
+        await TestWithAllCodeStyleOptionsOffAsync(
+            """
+            using System;
+            using System.Collections;
+            using System.Collections.Generic;
+
+            class Class : {|CS0535:{|CS0535:{|CS0535:{|CS0535:{|CS0535:IEnumerator<int>|}|}|}|}|}
+            {
+            }
+            """,
+            """
+            using System;
+            using System.Collections;
+            using System.Collections.Generic;
+
+            class Class : IEnumerator<int>
+            {
+                public int Current
+                {
+                    get
+                    {
+                        throw new NotImplementedException();
+                    }
+                }
+            
+                object IEnumerator.Current
+                {
+                    get
+                    {
+                        return Current;
```

Comment:
>return Current; [](http://example.com/codeflow?start=24&length=15)

Is this desired? The boxing might not be necessary depending on how the Current method is implemented.

Reply:
This is "fine".  it's the legacy apis calling to the modern ones.